src/pymor/algorithms/eigs.py

```diff
@@ -0,0 +1,292 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2020 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+import scipy.linalg as spla
+
+from pymor.algorithms.gram_schmidt import gram_schmidt
+from pymor.operators.constructions import IdentityOperator
+
+
+def eigs(A, E=None, k=3, which='LM', b=None, l=None, maxiter=1000, tol=1e-13):
+    """Approximate a few eigenvalues of an |Operator|.
+
+    Computes `k` eigenvalues `w[i]` with corresponding eigenvectors `v[i]` which solve
+    the eigenvalue problem
+
+    .. math::
+        A v[i] = w[i] v[i]
+
+    or the generalized eigenvalue problem
+
+    .. math::
+        A v[i] = w[i] E v[i]
+
+    if `E` is not `None`.
+
+    The implementation is based on Algorithm 4.2 in [RL95]_.
+
+    Parameters
+    ----------
+    A
+        The real |Operator| for which the eigenvalues are to be computed.
+    E
+        The |Operator| which defines the generalized eigenvalue problem.
+    k
+        The number of eigenvalues and eigenvectors which are to be computed.
+    which
+        A string specifying which `k` eigenvalues and eigenvectors to compute:
+            - `'LM'`: select eigenvalues with largest |v[i]|
+            - `'SM'`: select eigenvalues with smallest |v[i]|
+            - `'LR'`: select eigenvalues with largest Re(v[i])
+            - `'SR'`: select eigenvalues with smallest Re(v[i])
+            - `'LI'`: select eigenvalues with largest Im(v[i])
+            - `'SI'`: select eigenvalues with smallest Im(v[i])
+    b
+        Initial vector for Arnoldi iteration. Default is a random vector.
+    l
+        The size of the Arnoldi factorization. Default is `min(n - 1, max(2*k + 1, 20))`.
+    maxiter
+        The maximum number of iterations.
+    tol
+        The relative error tolerance for the ritz estimates.
+
+    Returns
+    -------
+    w
+        A |NumPy array| which contains the computed eigenvalues.
+    v
+        A |VectorArray| which contains the computed eigenvectors.
+    """
+
+    n = A.source.dim
+
+    if l is None:
+        l = np.min((n - 1, np.max((2 * k + 1, 20))))
```

Comment:
The built-in `min` and `max` are fine here (probably better than NumPy's, since it has to build the arrays).
```suggestion
        l = min(n - 1, max(2 * k + 1, 20))
```